src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp

```diff
@@ -189,23 +193,143 @@ bool ConvHipImplicitGemmGroupFwdXdlops::CheckCKApplicability(
+        }
+    }
+}
+
+bool PerformanceConfigHipImplicitGemmGroupFwdXdlops::ModelApplyToken(int idx, std::string value)
+{
+
+    std::vector<int> new_heuristic_indexes;
+    new_heuristic_indexes.reserve(heuristic_indexes.size());
+    if(idx >= 5)
+        idx += 2; // skip MPerXDL and NPerXDL as they are constant
+    std::copy_if(
+        heuristic_indexes.begin(),
+        heuristic_indexes.end(),
+        std::back_inserter(new_heuristic_indexes),
+        [&](int heuristic_index) { return heuristic_kernels[heuristic_index][idx] == value; });
+    if(new_heuristic_indexes.empty())
+        return false;
+    heuristic_indexes = new_heuristic_indexes;
+    return true;
+}
+
+static std::vector<float> GetFeatures(const ProblemDescription& problem, std::size_t num_cu)
+{
+    std::size_t n = 18;
+    std::vector<float> features(n, 0.0f);
+    features[0]  = problem.GetInDataType() == miopenFloat ? 2 : 1;
+    features[1]  = problem.GetInChannels_();
+    features[2]  = problem.GetInHeight_();
+    features[3]  = problem.GetInWidth_();
+    features[4]  = problem.GetOutChannels_();
+    features[5]  = problem.GetOutHeight_();
+    features[6]  = problem.GetOutWidth_();
+    features[7]  = problem.GetWeightsHeight_();
+    features[8]  = problem.GetWeightsWidth_();
+    features[9]  = problem.GetPadH();
+    features[10] = problem.GetPadW();
+    features[11] = problem.GetKernelStrideH();
+    features[12] = problem.GetKernelStrideW();
+    features[13] = problem.GetDilationH();
+    features[14] = problem.GetDilationW();
+    features[15] = problem.GetBatchSize_();
+    features[16] = problem.GetGroupCount();
+    features[17] = num_cu;
+    return features;
+}
+
+template <typename DataType>
+void PerformanceConfigHipImplicitGemmGroupFwdXdlops::RunParameterPredictionModel(
+    const ExecutionContext& ctx, const ProblemDescription& problem)
+{
+    valid_kernels = FillValidKernelsIDs<DeviceOpGFwdPtrs<DataType>, CKArgs>(
+        problem); // filter valid_kernel ID's
+    InitHeuristicKernelIDs();
+    static const std::string& arch  = ctx.GetStream().GetDeviceName();
+    static const std::string solver = "ConvHipIgemmGroupFwdXdlops";
+    std::vector<float> features     = GetFeatures(problem, ctx.GetStream().GetMaxComputeUnits());
+    if(ai::tuning::ModelSetParams(arch, solver, features, false, [&](int idx, std::string value) {
+           return this->ModelApplyToken(idx, value);
+       }))
+    {
+        index     = heuristic_indexes[0];
+        kernel_id = valid_kernels[index];
+        MIOPEN_LOG_I("Params set by AI: " << ToString());
+    }
+    else
+    {
+        index     = 0;
+        kernel_id = valid_kernels[index];
+    }
+}
+
+bool PerformanceConfigHipImplicitGemmGroupFwdXdlops::IsModelApplicable(
+    const ExecutionContext& ctx, const ProblemDescription& problem) const
+{
+    if(ctx.GetStream().GetDeviceName() != "gfx90a")
+        return false;
```

Comment:
We validate the device name on each run. Can we just do this check once, and store the result in a static variable? Since you have a check later on that makes sure that MIOpen throws an error if the device changes

Reply:
When we have a gfx908, gfx940 models sure. Otherwise this would try and find a .ktn that would not exist. for example if this was run on gfx908 it would try to find gfx908_ConvHipIgemmGroupFwdXdlops_decoder.ktn.model file even though it doesn't exist atm.